### PR TITLE
Support linking to section headings in same tab

### PR DESCRIPTION
**Pull request by Blacksmoke16**
Allows doing like:

```md
# # Main Heading
# 
# Some text
# 
# ## Sub Heading
#
# [Back to top](#main-heading)
```

The `Back to top` link will now take you to the `Main Heading` heading in same same tab.  Old behavior was opening that section in a new tab.  Links to sections in other files, or external links, will still open a new tab.

### CODE REVIEW

File: src/compiler/crystal/tools/doc/markdown/parser.cr [hunk 1]
```diff
@@ -391,8 +391,7 @@ class Crystal::Doc::Markdown::Parser
         end
       when '['
         unless in_link
-          link = check_link str, (pos + 1), bytesize
-          if link
+          if link = check_link str, (pos + 1), bytesize
```
Comment (straight-shoota) [inline src/compiler/crystal/tools/doc/markdown/parser.cr:394]: This is change is unrelated unncessary.

Comment (Blacksmoke16) [reply src/compiler/crystal/tools/doc/markdown/parser.cr:394]: I can revert it if you want...

**Review by jhass — APPROVED** [review]

**Comment by straight-shoota** [comment]
I'd suggest to remove `target="_blank"` entirely. Reasons: https://css-tricks.com/use-target_blank/

**Comment by Blacksmoke16** [comment]
@straight-shoota Yea that works for me, done.

File: src/compiler/crystal/tools/doc/markdown/parser.cr (hunk 1 above)
Comment (oprypin) [reply src/compiler/crystal/tools/doc/markdown/parser.cr:394]: Well, now it's even more unrelated

**Review by straight-shoota — APPROVED** [review]

**Comment by Blacksmoke16** [comment]
Good to merge?
